### ISSUE TITLE
cant get 0.58 acc

### ISSUE THREAD

**Issue by nickhuangxinyu** [yumoxu/stocknet-code#1]
Hi, yumoxu.

great work about your paper, but i tried to reproduce the results you mentioned in your paper.

only 0.53 i can get.

Is there any tricks i should know to get 0.58?

Thanks

**Comment by thtang** [comment]
I have the same question !

**Comment by Chih-Ling-Hsu** [comment]
I got acc: 0.574405 without doing anything special.
Not 0.58 acc still, but I guess we can get 0.58 with more trials?

**Comment by crazyofapple** [comment]
@Chih-Ling-Hsu I wonder what things you did. I run the code with GTX 1080Ti and default config,yml, only get 0.435516 in the test set.

**Comment by Chih-Ling-Hsu** [comment]
I also run the code with GTX 1080Ti and default config.yml, following the instructions in README file.
I have tried a second time and still get 0.57 acc so I guess it was not just a rare case.

**Comment by p-null** [comment]
@Chih-Ling-Hsu  
Have you checked  the majority of the stock movement of the pre-processed data?

**Comment by LifangD** [comment]
Hi, guys~ Questions about predicting y with posterior z. To predict the last T day, I think the gold label is leaked into **_z_post_** then **_g._**.   
![image](https://user-images.githubusercontent.com/33337947/67372482-8f601a00-f5b0-11e9-808c-e785cbed29fe.png)
By the way, I implement code on pytorch with the posterior z to predict. The acc is 0.5564, and the mcc is 0.0949.

**Comment by hjc5858** [comment]
不管怎么调，结果每次都是 0.5744。有问题！

**Comment by bahung** [comment]
> @Chih-Ling-Hsu I wonder what things you did. I run the code with GTX 1080Ti and default config,yml, only get 0.435516 in the test set.

Mine is 0.483135 not much different both in py27 and py36 with CUDA10

**Comment by bahung** [comment]
> I also run the code with GTX 1080Ti and default config.yml, following the instructions in README file.
> I have tried a second time and still get 0.57 acc so I guess it was not just a rare case.

I tried with Tesla P100 and got the same result with you by 30 epochs.

**Comment by zqy1106** [comment]
i have the same question

**Comment by WJMacro** [comment]
> > @Chih-Ling-Hsu I wonder what things you did. I run the code with GTX 1080Ti and default config,yml, only get 0.435516 in the test set.
> 
> Mine is 0.483135 not much different both in py27 and py36 with CUDA10

Hi, I'm running this repo with CUDA10. It takes a long time to finish one epoch. Did this happen in your test?

**Comment by WJMacro** [comment]
> I got acc: 0.574405 without doing anything special.
> Not 0.58 acc still, but I guess we can get 0.58 with more trials?

The test dataset has 1008 samples, 579 of which are positive.
0.574405=579/1008.
Maybe you should check your confusion matrix to see whether all samples in the testset were predicted to be positive.

**Comment by sia-watsonlee** [comment]
does anyone here have this implementation on pytorch or tensorflow2 ? i'm a beginner i want to run this on colab but it doesn't work since it required tensorflow 2

**Comment by Sia2010** [comment]
> 不管怎么调，结果每次都是 0.5744。有问题！

请问vocab.txt是什么呢，怎么下载呀

**Comment by Sia2010** [comment]
> Hi, yumoxu.
> 
> great work about your paper, but i tried to reproduce the results you mentioned in your paper.
> 
> only 0.53 i can get.
> 
> Is there any tricks i should know to get 0.58?
> 
> Thanks

Hi, could you plz tell me what the vocab.txt refer to? Where can I download it?

**Comment by kkew3** [comment]
Running on cpu, with default config and exactly the same dependency versions as in README, on test set, mine is 0.42 for 15 epochs (loss = 1.06), and 0.45 for 30 epochs (loss = 1.54). Interestingly, it goes up to 0.53 for one epoch (loss = 2.35).
